pkg/operator/hostetcdendpointcontroller/hostendpointcontroller.go

```diff
@@ -143,13 +143,15 @@ func (h *HostEtcdEndpointController) sync() error {
 		return fmt.Errorf("unexpected length of host endpoint subset")
 	}
 
```

Comment:
We need to revisit this method its very hard to follow what is going on and it seems like we should be able to do this simpler but for the sake of time it works for me.

Reply:
sure thanks, I would love to see how to simplify this. 

There are a bunch of corner cases I am taking care of so we don't break host-endpoint. It is critical to KAS calculating storage URLs so being extra careful